src/app/routes/app-routes.tsx

```diff
@@ -82,144 +87,172 @@ export const homePageModalRoutes = (
+            }
+          />
+          <Route
+            path={RouteUrls.AddNetwork}
+            element={
+              <AccountGate>
+                <AddNetwork />
+              </AccountGate>
+            }
+          />
+          <Route
+            path={RouteUrls.ChooseAccount}
+            element={
+              <AccountGate>
+                <ChooseAccount />
+              </AccountGate>
+            }
+          >
+            {ledgerJwtSigningRoutes}
+          </Route>
 
-        <Route
-          path={RouteUrls.SignIn}
-          element={
-            <OnboardingGate>
-              <SignIn />
-            </OnboardingGate>
-          }
-        />
-        <Route
-          path={RouteUrls.AddNetwork}
-          element={
-            <AccountGate>
-              <AddNetwork />
-            </AccountGate>
-          }
-        />
-        <Route
-          path={RouteUrls.ChooseAccount}
-          element={
-            <AccountGate>
-              <ChooseAccount />
-            </AccountGate>
-          }
-        >
-          {ledgerJwtSigningRoutes}
-        </Route>
+          <Route
+            path={RouteUrls.Fund}
+            element={
+              <AccountGate>
+                <FundPage />
+              </AccountGate>
+            }
+          >
+            <Route path={RouteUrls.ReceiveStx} element={<ReceiveStxModal />} />
+            <Route path={RouteUrls.ReceiveBtc} element={<ReceiveBtcModal />} />
+          </Route>
+          <Route
+            path={RouteUrls.FundChooseCurrency}
+            element={
+              <AccountGate>
+                <ChooseCryptoAssetToFund />
+              </AccountGate>
+            }
+          >
+            <Route path={RouteUrls.ReceiveStx} element={<ReceiveStxModal />} />
+          </Route>
 
-        <Route
-          path={RouteUrls.Fund}
-          element={
-            <AccountGate>
-              <FundPage />
-            </AccountGate>
-          }
-        >
-          <Route path={RouteUrls.ReceiveStx} element={<ReceiveStxModal />} />
-          <Route path={RouteUrls.ReceiveBtc} element={<ReceiveBtcModal />} />
-        </Route>
-        <Route
-          path={RouteUrls.FundChooseCurrency}
-          element={
-            <AccountGate>
-              <ChooseCryptoAssetToFund />
-            </AccountGate>
-          }
-        >
-          <Route path={RouteUrls.ReceiveStx} element={<ReceiveStxModal />} />
-        </Route>
+          {sendCryptoAssetFormRoutes}
 
-        {sendCryptoAssetFormRoutes}
+          <Route
+            path={RouteUrls.ViewSecretKey}
+            element={
+              <AccountGate>
+                <ViewSecretKey />
+              </AccountGate>
+            }
+          />
+          <Route path={RouteUrls.Unlock} element={<Unlock />}>
+            {leatherIntroDialogRoutes}
+          </Route>
+        </Route>
 
         <Route
-          path={RouteUrls.ViewSecretKey}
+          path="approver"
```

Comment:
This is your `Approver` but outside of the generic `<Container`. 

In this file I am wrapping it with `ContainerLayout` so the `Header` is wider to show how we could do this. 

A good fit for this approach would be the `OnBoarding` pages. 

For `Approver` I think we shouldn't wrap it but instead load the `Header` inside that component as I have done [here](src/app/features/approver/approver-demo.tsx)

By grouping routes better I believe we can simplify our logic and remove all the pathname switches I added to get us away from storing headers in state.